mariadb-galera: single node cluster doesn't recover

<!--
 Before you open the bug report please review the following troubleshooting guide:

 - [Troubleshoot Bitnami Helm Chart Issues](https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues)
 -->

**Which chart**:
mariadb-galera 6.0.5

**Describe the bug**
When running the chart with `replicaCount: 1`, the corresponding pod doesn't come back up if the underlying node fails or it needs to be resheduled for some other reason.

Tail log of the failed pod:
```
2021-12-07 10:36:01 0 [Note] WSREP: Recovering GCache ring buffer: version: 2, UUID: cd9f46f5-5747-11ec-bf53-e64aee8f74a8, offset: 1280
2021-12-07 10:36:01 0 [Note] WSREP: GCache::RingBuffer initial scan... 0.0% ( 0/134217752 bytes) complete.
2021-12-07 10:36:01 0 [Note] WSREP: GCache::RingBuffer initial scan...100.0% (134217752/134217752 bytes) complete.
2021-12-07 10:36:01 0 [Note] WSREP: Recovering GCache ring buffer: found gapless sequence 1-13
2021-12-07 10:36:01 0 [Note] WSREP: GCache::RingBuffer unused buffers scan... 0.0% ( 0/6008 bytes) complete.
2021-12-07 10:36:01 0 [Note] WSREP: GCache::RingBuffer unused buffers scan...100.0% (6008/6008 bytes) complete.
2021-12-07 10:36:01 0 [Note] WSREP: GCache DEBUG: RingBuffer::recover(): found 0/13 locked buffers
2021-12-07 10:36:01 0 [Note] WSREP: GCache DEBUG: RingBuffer::recover(): free space: 134211720/134217728
2021-12-07 10:36:01 0 [Note] WSREP: Passing config to GCS: base_dir = /bitnami/mariadb/data/; base_host = 10.244.3.37; base_port = 4567; cert.log_conflicts = no; cert.optimistic_pa = yes; debug = no; evs.auto_evict = 0; evs.delay_margin = PT1S; evs.delayed_keep_period = PT30S; evs.inactive_check_period = PT0.5S; evs.inactive_timeout = PT15S; evs.join_retrans_period = PT1S; evs.max_install_timeouts = 3; evs.send_window = 4; evs.stats_report_period = PT1M; evs.suspect_timeout = PT5S; evs.user_send_window = 2; evs.view_forget_timeout = PT24H; gcache.dir = /bitnami/mariadb/data/; gcache.keep_pages_size = 0; gcache.mem_size = 0; gcache.name = galera.cache; gcache.page_size = 128M; gcache.recover = yes; gcache.size = 128M; gcomm.thread_prio = ; gcs.fc_debug = 0; gcs.fc_factor = 1.0; gcs.fc_limit = 16; gcs.fc_master_slave = no; gcs.max_packet_size = 64500; gcs.max_throttle = 0.25; gcs.recv_q_hard_limit = 9223372036854775807; gcs.recv_q_soft_limit = 0.25; gcs.sync_donor = no; gmcast.segment = 0; gmcast.version = 0; pc.announce_timeout = PT3S; p
2021-12-07 10:36:01 0 [Note] WSREP: Start replication
2021-12-07 10:36:01 0 [Note] WSREP: Connecting with bootstrap option: 0
2021-12-07 10:36:01 0 [Note] WSREP: Setting GCS initial position to 00000000-0000-0000-0000-000000000000:-1
2021-12-07 10:36:01 0 [Note] WSREP: protonet asio version 0
2021-12-07 10:36:01 0 [Note] WSREP: Using CRC-32C for message checksums.
2021-12-07 10:36:01 0 [Note] WSREP: backend: asio
2021-12-07 10:36:01 0 [Note] WSREP: gcomm thread scheduling priority set to other:0
2021-12-07 10:36:01 0 [Warning] WSREP: access file(/bitnami/mariadb/data//gvwstate.dat) failed(No such file or directory)
2021-12-07 10:36:01 0 [Note] WSREP: restore pc from disk failed
2021-12-07 10:36:01 0 [Note] WSREP: GMCast version 0
2021-12-07 10:36:01 0 [Warning] WSREP: Failed to resolve tcp://test-mariadb-galera-headless.test-mariadb.svc.cluster.local:4567
2021-12-07 10:36:01 0 [Note] WSREP: (79224e3f-9952, 'tcp://0.0.0.0:4567') listening at tcp://0.0.0.0:4567
2021-12-07 10:36:01 0 [Note] WSREP: (79224e3f-9952, 'tcp://0.0.0.0:4567') multicast: , ttl: 1
2021-12-07 10:36:01 0 [Note] WSREP: EVS version 1
2021-12-07 10:36:01 0 [Note] WSREP: gcomm: connecting to group 'galera', peer 'test-mariadb-galera-headless.test-mariadb.svc.cluster.local:'
2021-12-07 10:36:01 0 [ERROR] WSREP: failed to open gcomm backend connection: 131: No address to connect (FATAL)
at /bitnami/blacksmith-sandox/libgalera-26.4.9/gcomm/src/gmcast.cpp:connect_precheck():317
2021-12-07 10:36:01 0 [ERROR] WSREP: /bitnami/blacksmith-sandox/libgalera-26.4.9/gcs/src/gcs_core.cpp:gcs_core_open():220: Failed to open backend connection: -131 (State not recoverable)
2021-12-07 10:36:01 0 [ERROR] WSREP: /bitnami/blacksmith-sandox/libgalera-26.4.9/gcs/src/gcs.cpp:gcs_open():1633: Failed to open channel 'galera' at 'gcomm://test-mariadb-galera-headless.test-mariadb.svc.cluster.local': -131 (State not recoverable)
2021-12-07 10:36:01 0 [ERROR] WSREP: gcs connect failed: State not recoverable
2021-12-07 10:36:01 0 [ERROR] WSREP: wsrep::connect(gcomm://test-mariadb-galera-headless.test-mariadb.svc.cluster.local) failed: 7
2021-12-07 10:36:01 0 [ERROR] Aborting
```
For me, it seems like it can't resolve any IP of the headless service because that very pod is the only target of the service & not ready yet.

**To Reproduce**
Steps to reproduce the behavior:

1. Install the chart with `replicaCount: 1`
2. Remove the corresponding pod created by the statefulset, so that the statefulset spawns a new pod.
3. The new pods never becomes ready, it is stuck in an CrashLoopBackOff

**Expected behavior**
The (single-node) galera cluster should recover.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.7.1", GitCommit:"1d11fcb5d3f3bf00dbe6fe31b8412839a96b3dc4", GitTreeState:"clean", GoVersion:"go1.17.1"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"archive", BuildDate:"2021-10-28T06:55:39Z", GoVersion:"go1.17.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"32a137c122b0474c719988922410f4027a4b002e", GitTreeState:"clean", BuildDate:"2021-11-01T16:43:17Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
No additional context

Same Error here Tested to uninstall and remove all pvc still cat get a working cluster up.


`mariadb-mariadb-galera-0   0/1     Running   0          14s
mahe@provision:/opt/repos/k8s-onprem/service/keyclouk$ kubectl logs -f mariadb-mariadb-galera-0 -n keycloak
mariadb 13:40:13.46 
mariadb 13:40:13.46 Welcome to the Bitnami mariadb-galera container
mariadb 13:40:13.47 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb-galera
mariadb 13:40:13.47 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb-galera/issues
mariadb 13:40:13.47 
mariadb 13:40:13.48 INFO  ==> ** Starting MariaDB setup **
mariadb 13:40:13.52 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 13:40:13.61 INFO  ==> Initializing mariadb database
mariadb 13:40:13.65 WARN  ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable or does not exist. Configurations based on environment variables will not be applied for this file.
mariadb 13:40:29.45 INFO  ==> Starting mariadb in background
mahe@provision:/opt/repos/k8s-onprem/service/keyclouk$ kubectl logs -f mariadb-mariadb-galera-0 -n keycloak
mariadb 13:43:09.49 
mariadb 13:43:09.49 Welcome to the Bitnami mariadb-galera container
mariadb 13:43:09.50 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mariadb-galera
mariadb 13:43:09.50 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mariadb-galera/issues
mariadb 13:43:09.50 
mariadb 13:43:09.50 INFO  ==> ** Starting MariaDB setup **
mariadb 13:43:09.55 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 13:43:09.63 INFO  ==> Initializing mariadb database
mariadb 13:43:09.67 WARN  ==> The mariadb configuration file '/opt/bitnami/mariadb/conf/my.cnf' is not writable or does not exist. Configurations based on environment variables will not be applied for this file.
mariadb 13:43:09.68 INFO  ==> Persisted data detected. Restoring
mariadb 13:43:09.73 INFO  ==> ** MariaDB setup finished! **

mariadb 13:43:09.84 INFO  ==> ** Starting MariaDB **
mariadb 13:43:09.84 INFO  ==> Setting previous boot
2021-12-07 13:43:09 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (mysqld 10.5.13-MariaDB-log) starting as process 1 ...
2021-12-07 13:43:09 0 [Note] WSREP: Loading provider /opt/bitnami/mariadb/lib/libgalera_smm.so initial position: 00000000-0000-0000-0000-000000000000:-1
2021-12-07 13:43:09 0 [Note] WSREP: wsrep_load(): loading provider library '/opt/bitnami/mariadb/lib/libgalera_smm.so'
2021-12-07 13:43:09 0 [Note] WSREP: wsrep_load(): Galera 4.9(rXXXX) by Codership Oy <info@codership.com> loaded successfully.
2021-12-07 13:43:09 0 [Note] WSREP: CRC-32C: using 64-bit x86 acceleration.
2021-12-07 13:43:09 0 [Note] WSREP: Found saved state: 00000000-0000-0000-0000-000000000000:-1, safe_to_bootstrap: 1
2021-12-07 13:43:09 0 [Note] WSREP: GCache DEBUG: opened preamble:
Version: 2
UUID: 00000000-0000-0000-0000-000000000000
Seqno: -1 - -1
Offset: -1
Synced: 1
2021-12-07 13:43:09 0 [Note] WSREP: Skipped GCache ring buffer recovery: could not determine history UUID.
2021-12-07 13:43:09 0 [Note] WSREP: Passing config to GCS: base_dir = /bitnami/mariadb/data/; base_host = 10.0.0.240; base_port = 4567; cert.log_conflicts = no; cert.optimistic_pa = yes; debug = no; evs.auto_evict = 0; evs.delay_margin = PT1S; evs.delayed_keep_period = PT30S; evs.inactive_check_period = PT0.5S; evs.inactive_timeout = PT15S; evs.join_retrans_period = PT1S; evs.max_install_timeouts = 3; evs.send_window = 4; evs.stats_report_period = PT1M; evs.suspect_timeout = PT5S; evs.user_send_window = 2; evs.view_forget_timeout = PT24H; gcache.dir = /bitnami/mariadb/data/; gcache.keep_pages_size = 0; gcache.mem_size = 0; gcache.name = galera.cache; gcache.page_size = 128M; gcache.recover = yes; gcache.size = 128M; gcomm.thread_prio = ; gcs.fc_debug = 0; gcs.fc_factor = 1.0; gcs.fc_limit = 16; gcs.fc_master_slave = no; gcs.max_packet_size = 64500; gcs.max_throttle = 0.25; gcs.recv_q_hard_limit = 9223372036854775807; gcs.recv_q_soft_limit = 0.25; gcs.sync_donor = no; gmcast.segment = 0; gmcast.version = 0; pc.announce_timeout = PT3S; pc
2021-12-07 13:43:09 0 [Note] WSREP: Start replication
2021-12-07 13:43:09 0 [Note] WSREP: Connecting with bootstrap option: 0
2021-12-07 13:43:09 0 [Note] WSREP: Setting GCS initial position to 00000000-0000-0000-0000-000000000000:-1
2021-12-07 13:43:09 0 [Note] WSREP: protonet asio version 0
2021-12-07 13:43:09 0 [Note] WSREP: Using CRC-32C for message checksums.
2021-12-07 13:43:09 0 [Note] WSREP: backend: asio
2021-12-07 13:43:09 0 [Note] WSREP: gcomm thread scheduling priority set to other:0 
2021-12-07 13:43:09 0 [Warning] WSREP: access file(/bitnami/mariadb/data//gvwstate.dat) failed(No such file or directory)
2021-12-07 13:43:09 0 [Note] WSREP: restore pc from disk failed
2021-12-07 13:43:09 0 [Note] WSREP: GMCast version 0
2021-12-07 13:43:09 0 [Note] WSREP: (9d9f3eed-9158, 'tcp://0.0.0.0:4567') listening at tcp://0.0.0.0:4567
2021-12-07 13:43:09 0 [Note] WSREP: (9d9f3eed-9158, 'tcp://0.0.0.0:4567') multicast: , ttl: 1
2021-12-07 13:43:09 0 [Note] WSREP: EVS version 1
2021-12-07 13:43:09 0 [Note] WSREP: gcomm: connecting to group 'galera', peer 'mariadb-mariadb-galera-headless.keycloak.svc.cluster.local:'
2021-12-07 13:43:09 0 [Note] WSREP: (9d9f3eed-9158, 'tcp://0.0.0.0:4567') Found matching local endpoint for a connection, blacklisting address tcp://10.0.0.240:4567
2021-12-07 13:43:12 0 [Note] WSREP: EVS version upgrade 0 -> 1
2021-12-07 13:43:12 0 [Note] WSREP: PC protocol upgrade 0 -> 1
2021-12-07 13:43:12 0 [Warning] WSREP: no nodes coming from prim view, prim not possible
2021-12-07 13:43:12 0 [Note] WSREP: view(view_id(NON_PRIM,9d9f3eed-9158,1) memb {
	9d9f3eed-9158,0
} joined {
} left {
} partitioned {
})
2021-12-07 13:43:13 0 [Warning] WSREP: last inactive check more than PT1.5S ago (PT3.50129S), skipping check
2021-12-07 13:43:42 0 [Note] WSREP: PC protocol downgrade 1 -> 0
2021-12-07 13:43:42 0 [Note] WSREP: view((empty))
2021-12-07 13:43:42 0 [ERROR] WSREP: failed to open gcomm backend connection: 110: failed to reach primary view: 110 (Connection timed out)
	 at /bitnami/blacksmith-sandox/libgalera-26.4.9/gcomm/src/pc.cpp:connect():160
2021-12-07 13:43:42 0 [ERROR] WSREP: /bitnami/blacksmith-sandox/libgalera-26.4.9/gcs/src/gcs_core.cpp:gcs_core_open():220: Failed to open backend connection: -110 (Connection timed out)
2021-12-07 13:43:42 0 [ERROR] WSREP: /bitnami/blacksmith-sandox/libgalera-26.4.9/gcs/src/gcs.cpp:gcs_open():1633: Failed to open channel 'galera' at 'gcomm://mariadb-mariadb-galera-headless.keycloak.svc.cluster.local': -110 (Connection timed out)
2021-12-07 13:43:42 0 [ERROR] WSREP: gcs connect failed: Connection timed out
2021-12-07 13:43:42 0 [ERROR] WSREP: wsrep::connect(gcomm://mariadb-mariadb-galera-headless.keycloak.svc.cluster.local) failed: 7
2021-12-07 13:43:42 0 [ERROR] Aborting
mahe@provision:/opt/repos/k8s-onprem/service/keyclouk$`

--- a/bitnami/mariadb-galera/Chart.yaml
+++ b/bitnami/mariadb-galera/Chart.yaml
@@ -27,4 +27,4 @@ sources:
   - https://github.com/bitnami/bitnami-docker-mariadb-galera
   - https://github.com/prometheus/mysqld_exporter
   - https://mariadb.org
-version: 6.0.6
+version: 6.0.7

--- a/bitnami/mariadb-galera/templates/headless-svc.yaml
+++ b/bitnami/mariadb-galera/templates/headless-svc.yaml
@@ -19,4 +19,5 @@ spec:
     - name: sst
       port: 4444
       targetPort: sst
+  publishNotReadyAddresses: true
   selector: {{- include "common.labels.matchLabels" . | nindent 4 }}

